File: RB101_Programming_Foundations/Small Probems/Easy5/letter_swap.rb
#input : a string of words separated by spaces
#output : a string with first and last letter of the input string swapped

# split each word of the string into an array
# for each word, swap the first and last character
# join the word array back to a join with spaces


def swap(string)
  words = string.split
  swap_words = words.map do |word|
    word[0], word[-1] = word[-1], word[0]
    word
  end

  swap_words.join(' ')
end

puts swap('Oh what a wonderful day it is') == 'hO thaw a londerfuw yad ti si'
puts swap('Abcde') == 'ebcdA'
puts swap('a') == 'a'

File: RB101_Programming_Foundations/Small Probems/Easy5/Easy5_Python/letter_swap.py
# Given a string of words separated by spaces, write a method that takes this string of words and returns a string in which the first and last letters of every word are swapped.

# You may assume that every word contains at least one letter, and that the string will always contain at least one word. You may also assume that each string contains nothing but words and spaces


'''
input : string
output : a new string with every word's first and last letter swapped

Algo :
    split the string into array by ' '
    iterate through the array
        for each word, swap its first and last character
    join the array back by ' '
'''

def swap(string):
    array = string.split()
    swapped_array = [''] * len(array)

    for index, word in enumerate(array):
        word_array = list(word)
        word_array[0], word_array[-1] = word_array[-1], word_array[0]
        swapped_array[index] = ''.join(word_array)
    return ' '.join(swapped_array)




print(swap('Oh what a wonderful day it is') == 'hO thaw a londerfuw yad ti si')
print(swap('Abcde') == 'ebcdA')
print(swap('a') == 'a')